lib/lacerda/conversion/data_structure.rb

```diff
@@ -101,22 +100,62 @@ def add_properties_to_json_schema
         end
       end
 
-      def primitive_or_reference(type, is_required)
+      # returns the type of an array, given its specified type(s). This will be
+      # either exactly one basic type, or a oneOf in case there are 1+ types 
+      # or exactly 1 non-basic type. 
+      # As there are specied types in the array, `nil` should not be a valid value
```

Comment:
I don't understand that last sentence - you mean if required is not true, `nil` should be one of the `oneOf`s?